### ISSUE TITLE
cyclonedx-gomod "DBG dependency not found ...", produces almost empty BOM

### ISSUE THREAD

**Issue by TheDiveO** [CycloneDX/cyclonedx-gomod#76]
As suggested in this repository, I tried to reach out first via Stackoverflow [cyclonedx-gomod "DBG dependency not found ...", produces almost empty BOM](https://stackoverflow.com/questions/69329242/cyclonedx-gomod-dbg-dependency-not-found-produces-almost-empty-bom) but I got no responses there for some time. If I got the wrong part of the SO fleet, then please direct me to the correct part of the SO community. I noticed that there is no tag dedicated to cylconedx.

Anyway: I'm unsuccessfully trying to generate a BOM from a Go module using cyclonedx-gomod. I've downloaded a binary version from the project's repository https://github.com/CycloneDX/cyclonedx-gomod/releases/tag/v1.0.0-alpha.4, for ARM64 to use on a Raspi 4 with Ubuntu.

Now, when I run `cyclonedx-gomod app -json -output mymodule-bom.json -main cmd/myapp/main.go -licenses -std -verbose .` while inside my module, I see lots of DBG complaints about dependencies, such as:

```
dependency not found dependant=github.com/thediveo/lxkns@v0.20.6 dependency=github.com/spf13/cobra@v1.2.1
```

However, these are public repositories and they have already been used for building, so they're present in the Go module cache. The resulting BOM JSON file consists of maybe four or five entries, but is completely useless because it lacks all dependencies.

What am I doing wrong?

**Comment by nscuro** [comment]
Hi @TheDiveO, I saw your post on Stack Overflow and created #75. I stumbled upon it by accident, and I don't believe anyone of the CycloneDX team actively monitors SO. If you happen to run into issues, please raise them here.

Regarding your problems: You're doing nothing wrong, it's a bug in *cyclonedx-gomod*, see #75. 

The `dependency not found` **debug** message you're seeing is somewhat unrelated to the underlying issue. Admittedly, the message is a bit misleading. It just means that `github.com/spf13/cobra@v1.2.1` (as taken from `go mod graph`'s output) was not found in the list of modules required by your application or module.

This typically happens because `go mod graph` is very verbose. It shows dependencies to modules that are not actually needed by your module or application. For example, when running `cyclonedx-gomod app ...` you'll *always* get this log message for test dependencies like `github.com/stretchr/testify`.

In your case, *cyclonedx-gomod* failed to recognize all packages / modules required by your application. Thus the list of "selected modules" doesn't contain `github.com/spf13/cobra`.

Thanks for reporting, I'll work on a fix later today and will hopefully be able to publish it with `v1.0.0-beta.2`.
I'll also adjust the debug message to reflect more clearly what's actually happening.

**Comment by nscuro** [comment]
@TheDiveO A fix has just been published with [`v1.0.0-beta.2`](https://github.com/CycloneDX/cyclonedx-gomod/releases/tag/v1.0.0-beta.2). Could you please verify that it now works as expected from your end?

**Comment by TheDiveO** [comment]
@nscuro Just tested and this now works as expected: the bom gets populated with dependencies of the main module. Thank you very much!
